lib/commands/implementations/mutelinks.js

```diff
@@ -7,17 +7,26 @@ const parseDurationToSeconds = require('../../chat-utils/duration-parser');
 const { makeMute } = require('../../chat-utils/punishment-helpers');
 const formatDuration = require('../../chat-utils/format-duration');
 
+/**
+ * @param {number} defaultPunishmentDuration
+ */
 function mutelinks(defaultPunishmentDuration) {
-  // States: "on" | "off" | "all"
+  /**
+   * @type {("on"|"off"|"all")}
+   */
   let state = 'off';
   // The user to be mentioned, if null then don't care about mention
   let mentionUser = null;
 
   let muteDuration = null;
 
-  return (input, services, rawMessage) => {
+  /**
+   * @param {string} input
+   * @param {import("../../services/service-index")} services
+   */
+  function handler(input, services, rawMessage) {
```

Comment:
I'm afraid if creating a maintenance burden by documenting its type tbh, since `rawMessage` is used elsewhere. If you have suggestion on how to implement it so I can import it like with `services` I'm all ears.

Reply:
Yeah, I'm not sure if it's possible just using JSDoc looks like, and it's definitely not a good idea to copy that type all around for sure. Using native EventEmitters we also can't really adding typings to the listener anyways, so it's probably less important in general.

One thing we can do that would be good is to add and `@returns {EventEmitter}` for this function, so when you use that service the `listener` is at least typed to an event emitter (even though we can't really get types for the parameters).

Reply:
Return type can now be inferred in MessageRelay.